src/PoolManager.sol

```diff
@@ -47,21 +47,25 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
 
     constructor(uint256 controllerGasLimit) ProtocolFees(controllerGasLimit) {}
 
+    function _getPool(PoolId id) internal view override returns (Pool.State storage) {
+        return pools[id];
+    }
+
     /// @inheritdoc IPoolManager
     function getSlot0(PoolId id)
         external
         view
         override
         returns (uint160 sqrtPriceX96, int24 tick, uint24 protocolFee, uint24 swapFee)
     {
-        Pool.Slot0 memory slot0 = pools[id].slot0;
+        Pool.Slot0 memory slot0 = _getPool(id).slot0;
```

Comment:
Could still just use `pools[id]` throughout PoolManager instead of `_getPool(id)` -- saves about 20 gas in some areas but might not be worth it